config-vault/src/main/java/io/scalecube/config/vault/KubernetesVaultTokenSupplier.java

```diff
@@ -0,0 +1,30 @@
+package io.scalecube.config.vault;
+
+import com.bettercloud.vault.EnvironmentLoader;
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import io.scalecube.config.utils.ThrowableUtil;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class KubernetesVaultTokenSupplier implements VaultTokenSupplier {
+
+  private static final String VAULT_ROLE = "VAULT_ROLE";
+  private static final String SERVICE_ACCOUNT_TOKEN_PATH =
+      "/var/run/secrets/kubernetes.io/serviceaccount/token";
+
+  @Override
+  public String getToken(EnvironmentLoader environmentLoader, VaultConfig config) {
+    String role = Objects.requireNonNull(environmentLoader.loadVariable(VAULT_ROLE), "vault role");
+    try {
+      String jwt = Files.lines(Paths.get(SERVICE_ACCOUNT_TOKEN_PATH)).collect(Collectors.joining());
```

Comment:
instead of collect, `map(jwt -> new Vault(config).auth().loginByKubernetes(role, jwt).getAuthClientToken(),
￼          "vault token").findFirst().orElseThrow(IllegalArgumentException::new)`

Reply:
It reads a file completely, so in that case, the JWT is all lines in the given file.